lib/ignore/ignore.go

```diff
@@ -52,6 +60,27 @@ func (p Pattern) String() string {
 	return ret
 }
 
+func (p Pattern) allowsSkippingIgnoredDirs() bool {
+	if p.result.IsIgnored() {
+		return true
+	}
+	if p.pattern[0] != '/' {
+		return false
+	}
+	// Double asterisk everywhere in the path except at the end is bad
+	tocheck := strings.TrimSuffix(p.pattern[1:], "**")
```

Comment:
why do we [1:] it here?

Reply:
No good reason anymore - remnant from iterating over all path components.